src/List.mo

```diff
@@ -292,7 +295,12 @@ module {
   ///
   /// Runtime: `O(size)`
   public func map<T, R>(list : List<T>, f : T -> R) : List<R> {
-    let blocks = VarArray.repeat<[var ?R]>([var], list.blocks.size());
+    let blocks = VarArray.repeat<[var ?R]>(
+      [var],
+      newIndexBlockLength(
+        Nat32.fromNat(if (list.elementIndex == 0) list.blockIndex - 1 else list.blockIndex)
```

Comment:
I don't understand this part: shouldn't we use this value from this line as `blocksCount` below?

If `blocksCount` is set to `list.blocks.size()` then we can overflow the index block below, or?